src/daycare.c

```diff
@@ -0,0 +1,1316 @@
+    GetBoxMonNick(&daycareMon->mon, gStringVar1);
+    return numLevelsGained;
+}
+
+static u32 GetDaycareCostForSelectedMon(struct DaycareMon *daycareMon)
+{
+    u32 cost;
+
+    u8 numLevelsGained = GetNumLevelsGainedFromSteps(daycareMon);
+    GetBoxMonNick(&daycareMon->mon, gStringVar1);
+    cost = 100 + 100 * numLevelsGained;
+    ConvertIntToDecimalStringN(gStringVar2, cost, STR_CONV_MODE_LEFT_ALIGN, 5);
+    return cost;
+}
+
+static u16 GetDaycareCostForMon(struct DayCare *daycare, u8 slotId)
+{
+    return GetDaycareCostForSelectedMon(&daycare->mons[slotId]);
+}
+
+void GetDaycareCost(void)
+{
+    gSpecialVar_0x8005 = GetDaycareCostForMon(&gSaveBlock1Ptr->daycare, gSpecialVar_0x8004);
+}
+
+static void Debug_AddDaycareSteps(u16 numSteps)
+{
+    gSaveBlock1Ptr->daycare.mons[0].steps += numSteps;
+    gSaveBlock1Ptr->daycare.mons[1].steps += numSteps;
+}
+
+u8 GetNumLevelsGainedFromDaycare(void)
+{
+    if (GetBoxMonData(&gSaveBlock1Ptr->daycare.mons[gSpecialVar_0x8004], MON_DATA_SPECIES) != 0)
+        return GetNumLevelsGainedForDaycareMon(&gSaveBlock1Ptr->daycare.mons[gSpecialVar_0x8004]);
+
+    return 0;
+}
+
+static void ClearDaycareMonMisc(struct DaycareMiscMon *misc)
+{
+    s32 i;
+
+    for (i = 0; i < OT_NAME_LENGTH + 1; i++)
+        misc->OT_name[i] = 0;
+    for (i = 0; i < POKEMON_NAME_LENGTH + 1; i++)
+        misc->monName[i] = 0;
+
+    ClearMailStruct(&misc->mail);
+}
+
+static void ClearDaycareMon(struct DaycareMon *daycareMon)
+{
+    ZeroBoxMonData(&daycareMon->mon);
+    daycareMon->steps = 0;
+    ClearDaycareMonMisc(&daycareMon->misc);
+}
+
+static void ClearAllDaycareData(struct DayCare *daycare)
+{
+    u8 i;
+
+    for (i = 0; i < DAYCARE_MON_COUNT; i++)
+        ClearDaycareMon(&daycare->mons[i]);
+
+    daycare->offspringPersonality = 0;
+    daycare->stepCounter = 0;
+}
+
+// Determines what the species of an Egg would be based on the given species.
+// It determines this by working backwards through the evolution chain of the
+// given species.
+static u16 GetEggSpecies(u16 species)
+{
+    int i, j, k;
+    bool8 found;
+
+    // Working backwards up to 5 times seems arbitrary, since the maximum number
+    // of times would only be 3 for 3-stage evolutions.
+    for (i = 0; i < EVOS_PER_MON; i++)
+    {
+        found = FALSE;
+        for (j = 1; j < NUM_SPECIES; j++)
+        {
+            for (k = 0; k < EVOS_PER_MON; k++)
+            {
+                if (gEvolutionTable[j].evolutions[k].targetSpecies == species)
+                {
+                    species = j;
+                    found = TRUE;
+                    break;
+                }
+            }
+
+            if (found)
+                break;
+        }
+
+        if (j == NUM_SPECIES)
+            break;
+    }
+
+    return species;
+}
+
+static s32 GetSlotToInheritNature(struct DayCare *daycare)
+{
+    u32 species[DAYCARE_MON_COUNT];
+    s32 i, slot = -1, dittoCount;
```

Comment:
If this formatting isn't necessary, please change it.

Reply:
This I can change.